core/components/awsMultiClient.go

```diff
@@ -0,0 +1,136 @@
+
+	config *aws.Config
+}
+
+// Configure method
+func (client *AwsMultiClient) Configure(conf core.PluginConfigReader) {
+	if client.endpoint == "" {
+		if client.region != defaultRegion {
+			client.endpoint = fmt.Sprintf("s3-%s.amazonaws.com", client.region)
+		} else {
+			client.endpoint = defaultEndpoint
+		}
+	}
+
+	client.config = aws.NewConfig()
+
+	client.config.WithRegion(client.region)
+	client.config.WithEndpoint(client.endpoint)
+
+	client.config.CredentialsChainVerboseErrors = aws.Bool(true)
+	client.config.WithCredentials(client.Credentials.CreateAwsCredentials())
+}
+
+// GetS3Client returns a s3.S3 client
+func (client *AwsMultiClient) GetS3Client() (*s3.S3, error) {
+	sess, err := client.newSessionWithOptions()
+	if err != nil {
+		return nil, err
+	}
+
+	return s3.New(sess, client.config), nil
+}
+
+func (client *AwsMultiClient) newSessionWithOptions() (*session.Session, error) {
+	return session.NewSessionWithOptions(session.Options{
+		Config:            *client.config,
+		SharedConfigState: session.SharedConfigEnable,
+	})
+}
+
+// AwsCredentials is a config struct for aws credential handling
+//
+// Credential/Type defines the credentials that are to be used when
+// connecting to s3. This can be one of the following: "environment",
+// "static", "shared" and "none". By default this is set to "none".
+// See https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/#Credentials for more information
+//
+// Credential/Id is used for "static" type and is used as the AccessKeyID
+//
+// Credential/Token is used for "static" type and is used as the SessionToken
+//
+// Credential/Secret is used for "static" type and is used as the SecretAccessKey
+//
+// Credential/File is used for "shared" type and is used as the path to your
+// shared Credentials file (~/.aws/credentials)
+//
+// Credential/Profile is used for "shared" type and is used for the profile
+//
+type AwsCredentials struct {
+	credentialType string `config:"Credential/Type" default:"none"`
+	staticID       string `config:"Credential/Id" default:""`
+	staticToken    string `config:"Credential/Token" default:""`
+	staticSecret   string `config:"Credential/Secret" default:""`
+	sharedFile     string `config:"Credential/File" default:""`
+	sharedProfile  string `config:"Credential/Profile" default:"default"`
+}
+
+// CreateAwsCredentials returns aws credentials.Credentials for active settings
+func (cred *AwsCredentials) CreateAwsCredentials() *credentials.Credentials {
+	switch cred.credentialType {
+	case credentialTypeEnv:
+		return credentials.NewEnvCredentials()
+
+	case credentialTypeStatic:
+		return credentials.NewStaticCredentials(cred.staticID, cred.staticSecret, cred.staticToken)
+
+	case credentialTypeShared:
+		return credentials.NewSharedCredentials(cred.sharedFile, cred.sharedProfile)
+
+	case credentialTypeNone:
+		return credentials.AnonymousCredentials
+
+	default:
+		panic(fmt.Sprintf("Unknown CredentialType: %s", cred.credentialType))
```

Comment:
return an error instead, please, so a plugin's configure method can in the end work with this.